domainobjs/ts/verifyingKey.ts

```diff
@@ -0,0 +1,159 @@
+import { G1Point, G2Point } from "maci-crypto"
+
+/**
+ * @notice A TS Class representing a zk-SNARK VerifyingKey
+ */
+export class VerifyingKey {
+	public alpha1: G1Point
+	public beta2: G2Point
+	public gamma2: G2Point
+	public delta2: G2Point
+	public ic: G1Point[]
+
+    /**
+     * Generate a new VerifyingKey
+     * @param _alpha1 the alpha1 point
+     * @param _beta2 the beta2 point
+     * @param _gamma2 the gamma2 point
+     * @param _delta2 the delta2 point
+     * @param _ic the ic points
+     */
+	constructor(
+		_alpha1: G1Point,
+		_beta2: G2Point,
+		_gamma2: G2Point,
+		_delta2: G2Point,
+		_ic: G1Point[]
+	) {
```

Comment:
I think it's better to have one object param. Usually if there are more that 3 params or these params have the same time, it's a sign to use one object instead. Not sure if it has compatibility issues but if not, it will be useful to implement it.

Reply:
What do you mean by "these params have the same time"? Is this pattern something you'd use in function signatures too, or only for classes constructors?

Reply:
Sorry, I mean `type`. This works for functions and constructors as well.

Reply:
Could you please share how you would rework this? would love to see this pattern in action and in other codebases too if you have any examples

Reply:
Sure.

```
interface IVerifyingKeyArgs {
  alpha1: G1Point;
  beta2: G2Point;
  gamma2: G2Point;
  delta2: G2Point;
  ic: G1Point[];
}

...

	constructor({
	  alpha1,
          beta2,
          gamma2,
          delta2,
          ic,
	}) {
          ...
        }
...
```

Reply:
I see, React props way 😃

Reply:
Let's keep this suggestion as a discussion item, and once we all agree on a way to go, we can update throughout the code?

Reply:
Ok, this works for me.